File: src/main/java/apache/ignite/futures/topicmessage/CancelAck.java
package apache.ignite.futures.topicmessage;

/** Cancellation confirmation sent from the server to the client. */
class CancelAck implements ServerResponse {
    /** Failure message. */
    private final String failure;

    /** Constructor. */
    CancelAck(String failure) {
        this.failure = failure;
    }

    /** @return Cancellation failure or {@code null} is cancellation succeeded. */
    String failure() {
        return failure;
    }
}

File: Apache.Ignite.Futures/TopicMessage/CancelAck.cs
namespace Apache.Ignite.Futures.TopicMessage
{
    /// <summary>
    /// Cancellation confirmation sent from the server to the client.
    /// </summary>
    internal class CancelAck
    {
        /// <summary>
        /// Gets or sets cancellation failure or <code>null</code> is cancellation succeeded.
        /// </summary>
        public string Failure { get; set; }
    }
}
